client/my-sites/stats/activity-log/index.jsx

```diff
@@ -14,14 +15,268 @@ import { getSiteSlug, isJetpackSite } from 'state/sites/selectors';
+			timestamp: 1483264820300
+		}
+	];
+
+	update_logs( log ) {
+		const { translate } = this.props;
+		switch ( log.type ) {
+			// Scans
+			case 'security_threat_found':
+				log.subTitle = translate( 'Security Threat Found!' );
+				log.icon = 'notice';
+				log.actionText = translate( 'Fix it' );
+			// Backups
+			case 'site_backed_up':
+				log.subTitle = translate( 'Site Backed up' );
+				log.icon = 'checkmark';
+				log.status = 'is-success';
+				log.actionText = 'Restore';
+				break;
+			// Backups
+			case 'site_backed_up_failed':
+				log.subTitle = translate( 'Site Backed up' );
+				log.icon = 'notice';
+				log.actionText = 'Fix';
+				log.status = 'is-error';
+				break;
+			// Plugins
+			case 'plugin_activated':
+				log.subTitle = translate( 'Plugin Activated' );
+				log.icon = 'plugins';
+				log.actionText = 'Deactivate';
+				break;
+			case 'plugin_deactivated':
+				log.subTitle = translate( 'Plugin Deactivated' );
+				log.icon = 'plugins';
+				log.actionText = 'Activate';
+				break;
+			case 'plugin_needs_update':
+				log.subTitle = translate( 'Plugin Update Available' );
+				log.icon = 'plugins';
+				log.status = 'is-warning';
+				log.actionText = 'Update';
+				break;
+			case 'suspicious_code':
+				log.subTitle = translate( 'Security Scan' );
+				log.icon = 'notice';
+				log.status = 'is-error';
+				log.actionText = 'Fix';
+				break;
+			case 'plugin_updated':
+				log.subTitle = translate( 'Plugin Updated' );
+				log.icon = 'plugins';
+				log.actionText = 'Revert';
+				break;
+
+			// Themes
+			case 'theme_switched':
+				log.subTitle = translate( 'Theme Activated' );
+				log.icon = 'themes';
+				log.actionText = 'Revert';
+				break;
+			case 'theme_updated':
+				log.subTitle = translate( 'Theme Updated' );
+				log.icon = 'themes';
+				log.actionText = 'Revert';
+				break;
+			// Plans
+			case 'plan_updated':
+				log.subTitle = translate( 'Plan' );
+				log.icon = 'clipboard';
+				log.actionText = 'Cancel';
+				break;
+			case 'plan_renewed':
+				log.subTitle = translate( 'Plan Renewed' );
+				log.icon = 'clipboard';
+				break;
+			// Jetpack Modules
+			case 'activate_jetpack_feature':
+				log.subTitle = translate( 'Jetpack' );
+				log.icon = 'cog';
+				log.actionText = 'Deactivate';
+				break;
+			case 'deactivate_jetpack_feature':
+				log.subTitle = translate( 'Jetpack' );
+				log.icon = 'cog';
+				log.actionText = 'Activate';
+				break;
+		}
+
+		return log;
+	}
+
 	render() {
-		const { slug, isJetpack } = this.props;
+		const { siteId, slug, isJetpack } = this.props;
+		const logs = this.logs();
+		const logsGroupedByDate = map(
+			groupBy(
+				logs.map( this.update_logs, this ),
+				( log ) => new Date( log.timestamp ).toDateString()
+			),
+			( daily_logs, timestamp ) => {
+				return (
+					<ActivityLogDay
+						key={ 'activity-log-' + timestamp }
+						logs={ daily_logs }
+						siteId={ siteId }
+						isRewindEnabled={ true }
+					/>
+				);
+			} );
```

Comment:
I'm inclined to use a more machine-type date than the human readable on produced by `toDateString`. Using the `toDateString`, `timestamp` is a bit misleading in the second function. I would propose something like this:

```js
const {
	isJetpack,
	moment,
	siteId,
	slug,
} = this.props;
const logs = this.logs();
const logsGroupedByDate = map(
	groupBy(
		logs.map( this.update_logs, this ),
		( log ) => moment( log.timestamp ).startOf( 'day' ).toISOString()
	),
	( daily_logs, isoString ) => (
		<ActivityLogDay
			key={ isoString }
			dateString={ moment( isoString ).format( 'LL' ) }
			logs={ daily_logs }
			siteId={ siteId }
			isRewindEnabled={ true }
		/>
	)
);
```

This also includes the suggestion about passing the `dateString` as a `prop`.